tests/Unit/TestHelpers.hpp

```diff
@@ -138,6 +138,52 @@ void test_move_semantics(T&& a, const T& comparison) {
 #pragma GCC diagnostic pop
 }
 
+// Test for iterators
+template <typename Container>
+void test_iterators(Container& c) {
+  CHECK(c.end() - c.begin() == c.size());
```

Comment:
A lot of stuff in here (such as the subtraction in this line) will only compile for random access iterators, but could be made to work for more using things like `std::distance` and `std::next`.  `std::set` (for the bidirectional case) and `std::unordered_set` (for the forward case) can be used as tests of what works.